aioshelly/rpc_device.py

```diff
@@ -0,0 +1,202 @@
+"""Shelly Gen2 RPC based device."""
+from __future__ import annotations
+
+from typing import Any, Dict
+
+import aiohttp
+
+from .common import ConnectionOptions, get_info
+from .exceptions import AuthRequired, NotInitialized
+from .wsrpc import WsRPC
+
+
+def mergedicts(dict1, dict2):
+    """Deep dicts merge."""
+    for k in set(dict1.keys()).union(dict2.keys()):
+        if k in dict1 and k in dict2:
+            if isinstance(dict1[k], dict) and isinstance(dict2[k], dict):
+                yield (k, dict(mergedicts(dict1[k], dict2[k])))
+            else:
+                yield (k, dict2[k])
+        elif k in dict1:
+            yield (k, dict1[k])
+        else:
+            yield (k, dict2[k])
```

Comment:
Seems that this is slightly simpler?

```suggestion
    result = dict(dict1)
    result.update(dict2)
    for key, value in result.items():
        if isinstance(value, dict) and isinstance(dict1.get(key), dict):
            result[key] = mergedicts2(dict1[key], value)
    return result
```